components/model/src/ome/model/internal/Permissions.java

```diff
@@ -300,6 +301,27 @@ public boolean isDisallowLink() {
         return isDisallow(restrictions, LINKRESTRICTION);
     }
 
+    public void addExtendedRestrictions(Set<String> extendedRestrictions) {
+        if (extendedRestrictions == null || extendedRestrictions.isEmpty()) {
+            return;
+        }
+
+        if (this.extendedRestrictions == null) {
+            this.extendedRestrictions = extendedRestrictions.toArray(
+                    new String[extendedRestrictions.size()]);
```

Comment:
It would be easier but less efficient, I think. Though now this is all getting complicated enough, that perhaps the few cycles would be worth it.